src/components/transactions/GroupLabel/index.tsx

```diff
@@ -1,43 +1,22 @@
-import { type ReactElement } from 'react'
+import type { ReactNode, ReactElement } from 'react'
 import type { Label } from '@gnosis.pm/safe-react-gateway-sdk'
 import { LabelValue } from '@gnosis.pm/safe-react-gateway-sdk'
 import css from './styles.module.css'
-import useTxQueue from '@/hooks/useTxQueue'
-import { isLabelListItem, isMultisigExecutionInfo, isTransactionListItem } from '@/utils/transaction-guards'
 import useSafeInfo from '@/hooks/useSafeInfo'
 
-export const useGroupLabel = (item: Label): string => {
-  const { page } = useTxQueue()
+export const useFutureNonceLabel = (label: LabelValue) => {
   const { safe } = useSafeInfo()
+  if (label !== LabelValue.Queued) return label
+  return `${label} - transaction with nonce ${safe.nonce} needs to be executed first`
+}
 
-  const { label } = item
-
-  if (label !== LabelValue.Queued || !page) {
-    return label
-  }
-
-  const firstTx = page.results.find(isTransactionListItem)
-
-  if (!firstTx || !isMultisigExecutionInfo(firstTx.transaction.executionInfo)) {
-    return label
-  }
-
-  const hasNextLabel = page.results.some((tx) => isLabelListItem(tx) && tx !== item)
-  const nonceInFuture = firstTx.transaction.executionInfo.nonce !== safe.nonce
-
-  if (!hasNextLabel && !nonceInFuture) {
-    return label
-  }
-
-  const nextNonce = nonceInFuture ? safe.nonce : firstTx.transaction.executionInfo.nonce
```

Comment:
This logic was added in #660. I replaced it with just always display the current safe nonce next to the "Queued" label.
It's possible that the actual next tx will have a nonce higher than the current safe none, but that's an edge case.